apps/01PigLatin.js

```diff
@@ -7,11 +7,128 @@ prompt.start();
+  //
+  //
+  // Since 'a' !== 'A' we would normally need search for the uppercase and lowercase strings.
+  //
+  // Since .toLowerCase makes every letter in a string lowercase we only need to search for each vowel one time.
+  //
+  //
+  word  = word.toLowerCase();
 
+  // Concept Check
+  //
+  //
+  // What is the significance of setting vowelIndex to -1?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+  //
+  // Your Answer:
+  //
+  // When searching  arrays a return value of -1 indicates that the value being searched for was not found.
+  //
+  //
+  // Initalizing vowelIndex to -1 gives vowelIndex the value of that would be returned if we did not find a vowel searching with 'indexOf'.
+  //
+  // If the search for a vowel completes and returns -1 then a vowel was not found.
+  //
+  // -1 helps reveal what is able to be found. If a search is performed and word.IndexOf returns an index > -1 then a vowel had been found.
+  //
+  //
+
+  var vowelIndex = -1 // Set it to assume there are no vowels
+
+  if ( ( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('a');
+  }
+
+  if ( ( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('e');
+  }
+
+  if ( ( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('i');
+  }
+
+  if ( ( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('o');
+  }
+
+  if ( ( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('u');
+  }
+
+  if ( ( word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('y');
+  }
+
+  // Concept Check
+  //
+  // Why are we using String.prototype.slice() here? What role does vowelIndex play?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice
+  //
+  // Your Answer:
+  //
+  // String.prototype.slice() extracts or cuts text from one string and with that text returns a new string. It starts to extract the string from the parameter beginSlice and stops at the index entered for the parameter endSlice.
+  //
+  // The index for endSlice' is not extracted in the new string. Any index with a value > endSlice are also not included in the new string.
+  //
+  // For example:
+  //
+  // var myName = 'mannyhagman';
+  // var mySlicedName = myName.slice(1, 5) ;
+  // will return a new string 'anny'
+  //
+  // We use slice() to extract parts of one string into two different new strings so they can be concatenated into 1 new string.
```

Comment:
Good answer to this! Just for future reference, in computer science it is commonly called a "substring." In other languages (like Java) it would be variable.substring() instead of .slice(). Javascript is just hipster that way.